elastic/beats
# [Filebeat] Update handling of elasticsearch server logs

# Status: in progress

- ✅  gc: no format change
- ✅  server
- ✅  slowlog
- ✅  deprecation
- ✅  audit
- ✅  Stack Monitoring end-to-end testing

## What does this PR do?

Fixes the elasticsearch module so that 8.0.0 ES json logs can be ingested.

`server`, `slowlog` and `deprecation` logs format did change in 8.0 and a dedicate pipeline was implemented for each log type.

`gc` kept the same format so we didn't have to update the ingestion pipeline for that. Not sure about `audit` yet.

<img width="1791" alt="Screenshot 2022-01-27 at 16 41 22" src="https://user-images.githubusercontent.com/5239883/151392230-dcbdc847-7322-4829-94d1-c7203c3f7942.png">

## Why is it important?

The current pipeline is unable to process elasticsearch JSON logs from 8.0.0. They just get silently dropped.

Missing these logs means we can't show relevant ES logs as part of the stack monitoring UI.

## Checklist


## Author's Checklist



## How to test this PR locally

Ensure you can run module tests locally. See https://github.com/elastic/beats/pull/30014 for reference.

Run `INTEGRATION_TESTS=1 BEAT_STRICT_PERMS=false TESTING_FILEBEAT_MODULES=elasticsearch ES_PASS=changeme pytest tests/system/test_modules.py -v` to test just the elasticsearch module.

Optionally:
- build filebeat and use it to ingest elasticsearch logs using the config found in https://github.com/elastic/beats/issues/29880
- add a kibana_system user (`curl -k -u elastic:changeme -H 'Content-Type: application/json'   http://localhost:9200/_security/user/kibana_system/_password -d'{"password": "changeme"}'`) to launch and inspect the logs in kibana

## Related issues

Fixes https://github.com/elastic/beats/issues/29880
